avaframe/com4FlowPy/flowCore.py

```diff
@@ -426,38 +423,37 @@ def calculation(args):
                 if (nodata in dem_ng) or np.size(dem_ng) < 9:
                     continue
 
-                if forestBool:
-                    cell_list.append(
-                        Cell(row[k], col[k],
-                             dem_ng, cellsize,
-                             flux[k], z_delta[k],
-                             cell,
-                             alpha, exp, flux_threshold, max_z_delta,
-                             startcell,
-                             FSI=forestArray[row[k], col[k]],
-                             forestParams=forestParams,
-                             )
-                    )
+                # if the current child cell is already in processedCells
+                # just add +1 to the visit-counter, else add it to the
+                # processedCells dictionary with visit-count = 1
+                if (row[k], col[k]) in processedCells:
+                    processedCells[(row[k], col[k])] += 1
                 else:
-                    cell_list.append(
-                        Cell(row[k], col[k],
-                             dem_ng, cellsize,
-                             flux[k], z_delta[k],
-                             cell,
-                             alpha, exp, flux_threshold, max_z_delta,
-                             startcell,
-                             )
-                    )
+                    processedCells[(row[k], col[k])] = 1
+
+                cell_list.append(Cell(
+                            row[k], col[k],
+                            dem_ng, cellsize,
+                            flux[k], z_delta[k],
+                            cell,
+                            alpha, exp, flux_threshold, max_z_delta,
+                            startcell,
+                            FSI=forestArray[row[k], col[k]] if isinstance(forestArray, np.ndarray) else None,
+                            forestParams=forestParams,
+                                     ))
 
             z_delta_array[cell.rowindex, cell.colindex] = max(z_delta_array[cell.rowindex, cell.colindex], cell.z_delta)
             flux_array[cell.rowindex, cell.colindex] = max(flux_array[cell.rowindex, cell.colindex], cell.flux)
-            count_array[cell.rowindex, cell.colindex] += int(1)
+
+            if processedCells[(cell.rowindex, cell.colindex)] == 1:
```

Comment:
why `` == 1`` ? 
``processedCells[(cell.rowindex, cell.colindex)]`` could also be larger, if the cell is hit more often (after lines 429, 430), so we don't consider these cells in the ``count_array``?